### DISCUSSION TITLE
Function and style of extended controls that conform to the object-oriented opening and closing principle

### DISCUSSION THREAD

**Discussion by isharrrry** [AvaloniaUI/Avalonia#9712] [General]
WPF supports inheriting a UI control. This subclass can inherit styles and extend functions and styles. However, AXAML does not support this practice. How to solve this problem

**Comment by maxkatz6** [comment]
> However, AXAML does not support this practice.

It does. Can you provide more information with samples?

I assume you are missing IStyleable.StyleKey property, as by default control is looking for styles/themes for `this.GetType` and not parent type:
```
public class MyButton : Button, IStyleable
{
    Type IStyleable.StyleKey => typeof(Button);
}
```

**Reply by robloo** [reply]
Also, don't forget that `ControlTheme` (which provides the "default style" with control template) was added in 11.0+. `ControlTheme` supports the `BasedOn` property like in WPF which allows you to modify/inherit "styles" (now called ControlTheme) for controls like WPF.
